### PR TITLE
Explorer: Improve Cardano Transaction Certification & Certificate Chain Validation UI

### LINKED ISSUE #1554: Explorer improvements Cardano transactions
## Why
Following the completion of issue #1497, we want to make some improvements on the explorer.

## What
Implement improvement in the Cardano transaction modal of the explorer.

## How
- [x] Add links to cexplorer for transactions
- [x] Verify transaction proof before validating the certificate chain
- [x] Display the beacon on which the verification takes place (latest at least of signed beacon in the artifact)
- [x] Use the Mithril icon for validated transactions?
- [x] Display more information about the certificates (epoch number, date of creation) in the verification modal
- [x] Add a link to the certificate modal in the verification modal?
- [x] Fix a display issue: when switching from an aggregator that have `cardano transactions` capabilities the tab container is empty.

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR includes several improvements to the explorer:
- Certificate Chain Validation Modal:
  - Validated certificated are now displayed in a table.
  - Improve error handling: If the validation failed from a wasm error the error is now displayed bellow the table.
image (1)

  - It now take it's `MithrilClient` and certificate as parameter, allowing parents to use them more easily and avoiding duplicate client instantiation.
  - It have a option to transform the certificate hash into links that, when clicked, raise a `onCertificateClick` event that you can wired to.
- Cardano Transaction Certification:
  - The proof are now verified before the certificate since their verification are independent and the proof verification is much faster.
  - Add the certificate beacon in the last step to tell the point of the Cardano chain at which the verification took place. 
  - Improve error handling: the "step breadcrumb" now show at which step the certification failed.
    - Note: if the error is raised in the "verifying proof" step then the certificate chain validation is skipped and it's tab can't be accessed.
  - The certificates hashes in the "Validating Certificate Chain" tab can now be clicked, this open the certificate details modal but with the "previous" and "verify" buttons removed (since you already have all the certificates in the table and verified the certificate chain).
- Signer registration page: the layout and names of the "epoch" navigation buttons have been redesigned to specify explicitly the two kinds of epoch they apply to (registration & current cardano).
image

- General:
  - Add sanchonet network to supported cexplorer list.
  - Add a machine generated "mithril-icons" font using [fantasticon](https://github.com/tancredi/fantasticon/) making integration and usage of custom icons far more easily. See explorer readme for more.
  - Reworked the import to use [absolute import](https://nextjs.org/docs/app/building-your-application/configuring/absolute-imports-and-module-aliases) (see `jsconfig.json` for the mapping).
  - Fix a regression related to the default aggregator capabilities that was throwing errors.
  - Fix the main tab becoming empty when switching the `Cardano Transactions` tab is selected and the use switch to another aggregator that doesn't support them. In that case the selected tab will switch to `Cardano Db`. 

## Demo
Showing most changes on the Cardano Transaction Certification and the Certicate Chain Validation:

[Screencast from 14-03-2024 10:12:15.webm](https://github.com/input-output-hk/mithril/assets/790521/c46903d2-e50c-409a-9d04-1eb8cbfa46a5)

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [x] Update README file (if relevant)

## Issue(s)
Closes #1554

### CODE REVIEW

File: mithril-explorer/src/components/CertifyCardanoTransactionsModal/TransactionCertificationResult.js [hunk 1]
```diff
@@ -35,12 +85,17 @@ export default function TransactionCertificationResult({
...
       ) : (
-        <Alert variant="danger" className="mb-0">
-          <Alert.Heading>Mithril could not certify the transactions</Alert.Heading>
+        <Alert variant="danger" className="mb-2">
+          <Alert.Heading>
+            <i className="text-danger bi bi-shield-slash"></i> Mithril could not certify the
+            transactions
+          </Alert.Heading>
           <p className="mb-1">
             Either the transactions proof is invalid or all the transactions are not certified.
```
Comment (jpraynaud) [inline mithril-explorer/src/components/CertifyCardanoTransactionsModal/TransactionCertificationResult.js:100]:
```suggestion
            Either the transactions proof is invalid or none of the transactions is certified.
```

File: mithril-explorer/README.md
Comment (jpraynaud) [inline mithril-explorer/README.md:1]: Is it worth explaining the unicode range to be used to add a new font entry?

**Review by jpraynaud — APPROVED** [review]
LGTM :rocket:

**Review by dlachaumepalo — APPROVED** [review]
LGTM 👍

**Review by sfauvel — APPROVED** [review]
Nothing to say. Good job.

**Pull request merged**